packages/block-library/src/comments-pagination/edit.js

```diff
@@ -63,6 +64,21 @@ export default function QueryPaginationEdit( {
 		],
 		__experimentalLayout: usedLayout,
 	} );
+
+	// Get the Discussion settings
+	const { pageComments } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const { __experimentalDiscussionSettings } = getSettings();
+		return __experimentalDiscussionSettings;
+	} );
```

Comment:
```suggestion
	const pageComments = useSelect( ( select ) => {
		const { getSettings } = select( blockEditorStore );
		const { __experimentalDiscussionSettings } = getSettings();
		return __experimentalDiscussionSettings?.pageComments;
	}, [] );
```

I would return only the data we need in the component. I doubt `__experimentalDiscussionSettings` will be updated often, but it's just good practice :)

Reply:
agreed, thanks for pointing it out and adding the dependencies array - missed it!